File: R-Name S/R-Name S_/z.Objective C/TransparentDraggedWindow.m
//
//  TransparentDraggedWindow.m
//  WindowTester
//
//  Created by Yoichi Tagaya on Wed May 01 2002.
//  Copyright (c) 2001 Yoichi Tagaya. All rights reserved.
//

#import "TransparentDraggedWindow.h"


@implementation TransparentDraggedWindow

- (id) initWithContentRect: (NSRect)contentRect
                 styleMask: (unsigned int)aStyle
                   backing: (NSBackingStoreType)bufferingType
                     defer: (BOOL)flag
{
   self = [super initWithContentRect: contentRect
                                    styleMask: aStyle
                                      backing: bufferingType
                                        defer: flag];
   return self;
}


- (unsigned int) draggingEntered: (id <NSDraggingInfo>)sender
{
   NSView *vwRoot = [self contentView];
   
   [[NSColor selectedControlColor] set];
   NSFrameRectWithWidth([vwRoot visibleRect], 2.5);
   [self displayIfNeeded];

   return NSDragOperationGeneric;
}

- (unsigned int) draggingUpdated: (id <NSDraggingInfo>)sender
{
   return NSDragOperationGeneric;
}

- (void) draggingExited: (id <NSDraggingInfo>)sender
{
   NSView *vwRoot = [self contentView];

   NSEraseRect([vwRoot visibleRect]);
   [self display];
}

- (BOOL) prepareForDragOperation: (id <NSDraggingInfo>)sender
{
   return YES;
}

- (BOOL) performDragOperation: (id <NSDraggingInfo>)sender
{
   if ([[self delegate] respondsToSelector: @selector(performDragToWindowOperation:)]) {
      return [[self delegate] performDragToWindowOperation: sender];
   }
   
   return NO;
}

- (void) concludeDragOperation: (id <NSDraggingInfo>)sender
{
   NSView *vwRoot = [self contentView];

   NSEraseRect([vwRoot visibleRect]);
   [self display];
}


@end

File: R-Name S old/Stubs/TransparentDraggedWindow.swift
//
//  TransparentDraggedWindow.swift
//  R-Name S
//
//  Created by Pedro Plowman on 14/5/19.
//  Copyright © 2019 p2made. All rights reserved.
//

import AppKit // needed?

class TransparentDraggedWindow {

	/*

	init(contentRect: NSRect, styleMask style: NSWindow.StyleMask, backing bufferingType: NSWindow.BackingStoreType, defer flag: Bool) {
		if let aStyle = NSWindow.StyleMask(rawValue: aStyle) {
			super.init(contentRect: contentRect, styleMask: aStyle, backing: bufferingType, defer: flag)
		}
	}

	func draggingEntered(_ sender: NSDraggingInfo) -> UInt {
		let vwRoot = contentView as? NSView

		NSColor.selectedControlColor.set()
		NSFrameRectWithWidth(vwRoot?.visibleRect, 2.5)
		displayIfNeeded()

		return NSDragOperation.generic.rawValue
	}

	func draggingUpdated(_ sender: NSDraggingInfo) -> UInt {
		return NSDragOperation.generic.rawValue
	}

	func draggingExited(_ sender: NSDraggingInfo?) {
		let vwRoot = contentView as? NSView

		NSEraseRect(vwRoot?.visibleRect)
		display()
	}

	func prepareForDragOperation(_ sender: NSDraggingInfo) -> Bool {
		return true
	}

	func performDragOperation(_ sender: NSDraggingInfo) -> Bool {
		if delegate().responds(to: #selector(TransparentDraggedWindow.performDrag(toWindowOperation:))) {
			return delegate().performDrag(toWindowOperation: sender)
		}

		return false
	}

	func concludeDragOperation(_ sender: NSDraggingInfo?) {
		let vwRoot = contentView as? NSView

		NSEraseRect(vwRoot?.visibleRect)
		display()
	}

	init(contentRect: NSRect, styleMask style: NSWindow.StyleMask, backing bufferingType: NSWindow.BackingStoreType, defer flag: Bool) {
		if let aStyle = NSWindow.StyleMask(rawValue: aStyle) {
			super.init(contentRect: contentRect, styleMask: aStyle, backing: bufferingType, defer: flag)
		}
	}

	func draggingEntered(_ sender: NSDraggingInfo) -> UInt {
		let vwRoot = contentView as? NSView

		NSColor.selectedControlColor.set()
		NSFrameRectWithWidth(vwRoot?.visibleRect, 2.5)
		displayIfNeeded()

		return NSDragOperation.generic.rawValue
	}

	func draggingUpdated(_ sender: NSDraggingInfo) -> UInt {
		return NSDragOperation.generic.rawValue
	}

	func draggingExited(_ sender: NSDraggingInfo?) {
		let vwRoot = contentView as? NSView

		NSEraseRect(vwRoot?.visibleRect)
		display()
	}

	func prepareForDragOperation(_ sender: NSDraggingInfo) -> Bool {
		return true
	}

	func performDragOperation(_ sender: NSDraggingInfo) -> Bool {
		if delegate().responds(to: #selector(TransparentDraggedWindow.performDrag(toWindowOperation:))) {
			return delegate().performDrag(toWindowOperation: sender)
		}

		return false
	}

	func concludeDragOperation(_ sender: NSDraggingInfo?) {
		let vwRoot = contentView as? NSView

		NSEraseRect(vwRoot?.visibleRect)
		display()
	}

	*/

}
